eliot/_message.py

```diff
@@ -89,6 +90,24 @@ def _timestamp(self):
         return self._time()
 
 
+    def _freeze(self, action=None):
```

Comment:
So as long as you're refactoring, how about changing internal data structure to be a `pyrsistent.PMap`?